src/app/app.component.ts

```diff
@@ -3,23 +3,56 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-import { Component, OnInit } from '@angular/core';
+import { Component, Inject, OnInit } from '@angular/core'
+import { ShareLink } from './shareUrl/sharelink-url'
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
 export class AppComponent implements OnInit {
+  readonly TRIPLE_BANANA_DEV_URL = "https://triplebanana.dev"
+
   readonly DEFAULT_BANNER_DESCRIPTION: string = "Safer, faster, and more stable"
+  readonly ANDROID_CHROME_CLICK_DESCRIPTION = "Click to open link with Banana browser"
+
+  readonly CLICK_GUIDE_WAITING_IN_MS = 500
+
   bannerDescription = this.DEFAULT_BANNER_DESCRIPTION
 
+  private shareLink: ShareLink
+
+  constructor(
+    @Inject('Location') private location: Location
+  ) {
+    this.shareLink = this.getShareLinkUrl()
+  }
+
   ngOnInit(): void {
-    // Parse parameters as url and move
+    this.processShareLink()
+  }
+
+  navigate(): void {
+    // Not Implemented
+  }
+
+  private processShareLink(): void {
```

Comment:
How about creating a new class that is responsible for handling URL like this logic?
Roughly, it seems like
```
export class ShareUrlHandler {
  public handleUrl(string url): void
}
```
the output would be navigating with a refined URL.
I think it would be testable for core logic regardless of an angular framework api like `ngOnInit`.

WDYT?